src/braket/circuits/synthesis/invariants.py

```diff
@@ -0,0 +1,90 @@
+# Copyright 2019-2019 Amazon.com, Inc. or its affiliates. All Rights Reserved.
+#
+# Licensed under the Apache License, Version 2.0 (the "License"). You
+# may not use this file except in compliance with the License. A copy of
+# the License is located at
+#
+#     http://aws.amazon.com/apache2.0/
+#
+# or in the "license" file accompanying this file. This file is
+# distributed on an "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF
+# ANY KIND, either express or implied. See the License for the specific
+# language governing permissions and limitations under the License.
+
+import numpy as np
+
+from braket.circuits.synthesis.constants import magic_basis
+from braket.circuits.synthesis.predicates import is_unitary
+from braket.circuits.synthesis.util import char_poly, to_su
+from braket.circuits.gates import Y
+
+# Pauli Y
+y = Y().to_matrix()
+
+
+def makhlin_invariants(
+    U: np.ndarray, validate_input: bool = True, atol: float = 1e-8, rtol: float = 1e-5
+) -> np.dtype:
+    """
+    Calculate the Makhlin invariants of a 2q gate.
+
+    Args:
+        U (np.ndarray): matrix to calculate the Makelin invariant.
+        validate_input (bool): if validate input.
+        atol (float): absolute tolerance of loss.
+        rtol (float): relative tolerance of loss.
+
+    Returns:
+        makhlin_invariants (np.dtype): the calculated Makhlin invariants.
```

Comment:
`np.dtype` seems to be the wrong return type. Looks to me like `Tuple[float, complex, float]`.